### PR TITLE
RouteFlat -> Resolved<Route>

### PR DESCRIPTION
**Pull request by stackoverfloweth**
- rename flat to resolved
 - using `Resolved<T>` approach to nesting types
 - include the rest of route in what was previously RouteFlat

### CODE REVIEW

File: src/types/resolved.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { Param } from '@/types/params'
+import { Route } from '@/types/routes'
 
-export type RouteFlat = {
+export type Resolved<T extends Route> = T & {
```
Comment (pleek91) [inline src/types/resolved.ts:4]: We could probably use `T` to tell us what `path` and `params` actually are. I don't think there's much of a benefit to `path` but maybe to `params`? No change requested. Just an idea.

Comment (pleek91) [reply src/types/resolved.ts:4]: On a separate note. Doesn't really seem like we're using `T` for anything. Should this just be `Route & { ... }`?

**Comment by pleek91** [comment]
I really like "Resolved Route" as a term over "RouteFlat". Seems more intuitive.

File: src/types/resolved.ts (hunk 1 above)
Comment (stackoverfloweth) [reply src/types/resolved.ts]: moved T into `matched` property, I still prefer having the explicitness of how this is written over `Resolved` just being a standalone type

**Review by pleek91 — APPROVED** [review]

**Pull request merged**